### ISSUE TITLE
Add support for super-resolution (16X) mode

### ISSUE THREAD

**Issue by oskay** [evil-mad/robopaint#88]
Low-priority

**Comment by docprofsky** [comment]
@techninja Would this need changes in both RoboPaint and CNCServer or just CNCServer?

**Comment by techninja** [comment]
CNC server only, it both. Honestly if we enabled it for the EBB/wcb, we'd have no reason to have lower resolution.

**Comment by docprofsky** [comment]
RoboPaint is supposed to work with many bots regardless of their coordinate system; so if changing the number of steps (and updating the dimensions correspondingly) breaks RP then that is a bug.

I will close this unless you can think of some reason we should have to change RoboPaint for this.

**Comment by techninja** [comment]
Yes, really this is just a bot setup thing for WCB, OR you could define another WCB as "super resolution WaterColorBot" then select it as an alternative machine.

**Comment by oskay** [comment]
Really, this is an option that could be configured in the machine definition file in cncserver.

Larger question: Should there be a clear support mechanism for machine-dependent options like that?

**Comment by techninja** [comment]
Yup, machine config is exactly what I was talking about :).

Machine dependent options is basically a CNCserver configuration that exposes machine specific config to client applications. We do that somewhat globally, but right now machine configs are entirely limited to ini files (and therefore can't really handle exposing custom configs). There's talk of _seriously_ extending machine config to far more easily allow for flexibility on what board type and separation of WCB specific stuff from CNC Server core (which I've wanted to do forever, and would make reinking/wash highly configurable). This method would allow for a completely pluggable render system (needed for controller boards that take entirely different command styles), AND custom exposing custom endpoints/config per bot.

I think I might be able to get this work started (and completed) this year, assuming i can get my last paid backburner project out of the way as it's over 3 months late now!
